pkg/resources/server.go

```diff
@@ -92,98 +156,48 @@ func detectPluginWatchMode(sockDir string) bool {
 	return true
 }
 
-// dial establishes the gRPC communication with the registered device plugin.
-func dial(unixSocketPath string, timeout time.Duration) (*grpc.ClientConn, error) {
-	var c *grpc.ClientConn
-	var err error
-	connChannel := make(chan interface{})
-	ctx, timeoutCancel := context.WithTimeout(context.TODO(), timeout)
-
-	defer timeoutCancel()
-
-	go func() {
-		c, err = grpc.DialContext(ctx, unixSocketPath, grpc.WithInsecure(), grpc.WithBlock(),
-			grpc.WithContextDialer(func(ctx context.Context, addr string) (net.Conn, error) {
-				return net.Dial("unix", addr)
-			}),
-		)
-		connChannel <- "done"
-	}()
-
-	select {
-	case <-ctx.Done():
-		return nil, fmt.Errorf("timout while trying to connect %s", unixSocketPath)
-
-	case <-connChannel:
-		return c, err
-	}
-}
-
-func getRdmaDeviceSpec(pciAddress string) []*pluginapi.DeviceSpec {
-	deviceSpec := make([]*pluginapi.DeviceSpec, 0)
-
-	rdmaDevices := utils.GetRdmaDevices(pciAddress)
-	for _, device := range rdmaDevices {
-		deviceSpec = append(deviceSpec, &pluginapi.DeviceSpec{
-			HostPath:      device,
-			ContainerPath: device,
-			Permissions:   "rwm"})
-	}
-
-	return deviceSpec
-}
-
 // Start starts the gRPC server of the device plugin
 func (m *resourceServer) Start() error {
-	err := m.cleanup()
-	if err != nil {
-		return err
-	}
+	_ = m.cleanup()
```

Comment:
what if cleanup fails ? if yes, should we not check the return value ?

Reply:
right, I will add the check

Reply:
I will add check for it